### PR TITLE
Refactor eula_policy action write path to be cleaner & safer

### PR DESCRIPTION
**Pull request by diox**
Fixes: mozilla/addons#14966

https://www.django-rest-framework.org/api-guide/viewsets/#routing-additional-http-methods-for-extra-actions

### CODE REVIEW

File: src/olympia/addons/tests/test_views.py [hunk 1]
```diff
@@ -3015,7 +3015,7 @@ def test_developer_version_serializer_used_for_authors(self):
         # not logged in
         assert 'source' not in self.client.get(self.url).data
 
-        user = UserProfile.objects.create(username='user')
+        user = user_factory()
```
Comment (diox) [inline src/olympia/addons/tests/test_views.py:3018]: drive-by, I replaced all `UserProfile.objects.create(username='user')` in this test file with `user_factory()` for consistency, which amongst other things ensures the user always has an email (which in turn matters for some of the eula/policy tests because of `isSubmissionAllowedFor` permission check).

File: src/olympia/addons/views.py [hunk 2]
```diff
@@ -346,20 +346,19 @@ def get_georestrictions(self):
...
-            kwargs['partial'] = True
-            method = self.update
-        return method(request, **kwargs)
+        return self.retrieve(request)
+
+    @eula_policy.mapping.patch
+    def update_eula_policy(self, request, pk=None):
+        kwargs = {'partial': True}
+        self.permission_classes = self.write_permission_classes
+        return self.update(request, **kwargs)
```
Comment (eviljeff) [inline src/olympia/addons/views.py:361]: nit, the kwargs dict seems unnecessary?
```suggestion
        self.permission_classes = self.write_permission_classes
        return self.update(request, partial=True)
```

**Review by eviljeff — APPROVED** [review]
